.github/workflows/check-for-build-warnings.yml

```
@@ -0,0 +1,10 @@
+on: [pull_request]
```

We've started making the following on other actions:

```suggestion
on: 
  pull_request:
    types: [opened, synchronize, reopened]
  workflow_dispatch:
    inputs:
      reason:
        description: 'The reason for running the workflow'
        required: true
        default: 'Manual run'
```

The first change runs the workflow only when a PR is opened, the branch is updated, or the PR is re-opened. That prevents it running when labels are added, or other PR events are triggered.

The second lets us run it manually (mostly for testing). The extra arguments require the person starting the manual process to state a reason for the run.